### PR TITLE
fix: Ensure normalize method reliably strips all trailing special chars

### PR DESCRIPTION
**Pull request by aulightzxy-web**
✨ Description
This PR fixes a bug in the ScriptService.normalize() method where it failed to reliably strip all invalid trailing characters (-, ., _).

The previous implementation used three separate, sequential calls to StringUtils.stripEnd(). This created a bug where a string ending in a combination of these characters, like "my-string-_", would be incorrectly processed. The final result would be "my-string-", which violates the RFC 1123 naming conventions that this method is intended to enforce.

The fix consolidates the three calls into a single, robust call: StringUtils.stripEnd(string, "-._"). This ensures that any combination of these invalid characters is correctly stripped from the end of the string.

The corresponding unit test for the normalize method has also been expanded to include assertions for these specific edge cases, verifying the fix and preventing future regressions.

🔗 Related Issue
N/A

🛠️ Backend Checklist
Code compiles successfully and passes all checks
All unit and integration tests pass

📝 Additional Notes
This is a correctness fix for a subtle bug in a core utility method.

🤖 AI Authors
What do you call a cat that gets everything it wants? Purr-suasive. 🐱

### CODE REVIEW

File: core/src/test/java/io/kestra/core/models/tasks/runners/ScriptServiceTest.java [hunk 1]
```diff
@@ -195,9 +195,19 @@ void jobName() {
 
     @Test
     void normalize() {
+        // existing tests
```
Comment (loicmathieu) [inline core/src/test/java/io/kestra/core/models/tasks/runners/ScriptServiceTest.java]:
```suggestion
```

This comment is useless

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks.

I'll push the suggestion before merging.

**Comment by loicmathieu** [comment]
Backported to 1.0, 1.1 and 1.2
